tests/test_backend_loki_field_modifiers.py

```diff
@@ -59,7 +59,7 @@ def loki_backend() -> LogQLBackend:
         "label_format match_0=`{{ if eq .fieldA .fieldA }}true{{ else }}false{{ end }}`"
         " | match_0=`true`",
     ),
-    "expand": ('"%test%"', "fieldA=~`(?i)valueA`"),
+    "expand": ('"%test%"', "fieldA=~`(?i)^valueA$`"),
```

Comment:
Shouldn't most of these `expand` (or be replaced) as is, instead of using a regexp?
```
fieldName=~`(?i)^something$`
```

Reply:
They are regular expressions because the test runs in case-insensitive mode (which makes sense, it is the default) and the only way of achieving case-insensitive searching in LogQL is via the `(?i)` regex flag.

Reply:
I was mostly curious about the start and end markers.

Reply:
This PR is solving [this issue](https://github.com/grafana/pySigma-backend-loki/issues/163), which explains that, when using a regular expression label filter, the strings are being treated as un-anchored regular expressions. This means that if you have an expand value `something` for `fieldName`, the query would match `fieldName=something` but also `fieldName=thisomethingelse` - which is not the expected behavior.

Reply:
Oh, that's why!